faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -320,19 +320,27 @@ private String mkUrl(String path) throws MalformedURLException {
     return new URL(faunaRoot, path).toString();
   }
 
-  private void logSuccess(Request request, Response response) throws IOException {
+  private void logSuccess(Request request, Response response) {
     if (log.isDebugEnabled()) {
       String data = Optional.fromNullable(request.getStringData()).or("");
       String host = Optional.fromNullable(response.getHeader(X_FAUNADB_HOST)).or("Unknown");
       String build = Optional.fromNullable(response.getHeader(X_FAUNADB_BUILD)).or("Unknown");
-      String body = Optional.fromNullable(response.getResponseBody()).or("");
+      String body = Optional.fromNullable(getResponseBody(response)).or("");
 
       log.debug(
         format("Request: %s %s: %s. Response: Status=%d, Fauna Host: %s, Fauna Build: %s: %s",
           request.getMethod(), request.getUrl(), data, response.getStatusCode(), host, build, body));
     }
   }
 
+  private String getResponseBody(Response response) {
+    try {
+      return response.getResponseBody();
+    } catch (IOException e) {
+      return null;
```

Comment:
In that case we don't even try to parse the body, it will be done in later stage with proper error handling. This method is called inside `logSuccess`